src/query/util/promqlengine/cache_test.go

```diff
@@ -48,23 +48,21 @@ func TestGet(t *testing.T) {
 	require.EqualError(t, err, time.Minute.String())
 }
 
-func TestSet(t *testing.T) {
-	cache := NewCache(func(l time.Duration) (*promql.Engine, error) {
-		return &promql.Engine{}, errors.New("not set")
-	})
-
+func TestDefaultEngines(t *testing.T) {
 	lookbacks := []time.Duration{time.Second, time.Minute, time.Hour}
-	expecteds := make([]*promql.Engine, 0)
+	expecteds := make(map[time.Duration]*promql.Engine)
 	for _, l := range lookbacks {
-		e := newTestEngine(l)
-		cache.Set(l, e)
-		expecteds = append(expecteds, e)
+		expecteds[l] = newTestEngine(l)
```

Comment:
Guess this might read cleaner if you drop `lookbacks` slice and construct `expecteds` map inline (no loop).